portal-web/test/functional/com/liferay/portalweb/macros/WorkflowMetrics.macro

```diff
@@ -11,6 +11,65 @@ definition {
 		}
 	}
 
+	macro applyRoleFilterToAssignee {
+		WorkflowMetrics.selectWorkloadByAssigneeCardFilterValue(workflowStep = "1Role");
+
+		Click(locator1 = "WorkflowMetrics#METRICS_WORKLOAD_BY_ASSIGNEE_SEARCH_ASSIGNEE_NAME");
+
+		WorkflowMetrics.viewWorkloadByAssignee(
+			assigneeName = "1user user1",
+			isNotPresent = "2user user2,3user user3,4user user4,5user user5,6user user6",
+			overdueTaskCount = "0",
+			onTimeTaskCount = "0",
+			totalPendingTaskCount = "1");
+
+		WorkflowMetrics.selectWorkloadByAssigneeCardFilterValue(workflowStep = "2Role");
+
+		Click(locator1 = "WorkflowMetrics#METRICS_WORKLOAD_BY_ASSIGNEE_SEARCH_ASSIGNEE_NAME");
+
+		WorkflowMetrics.viewWorkloadByAssignee(
+			assigneeName = "2user user2",
+			isNotPresent = "3user user3,4user user4,5user user5,6user user6",
+			overdueTaskCount = "0",
+			onTimeTaskCount = "0",
+			totalPendingTaskCount = "1");
+
+		WorkflowMetrics.selectWorkloadByAssigneeCardFilterValue(workflowStep = "3Role");
+
+		Click(locator1 = "WorkflowMetrics#METRICS_WORKLOAD_BY_ASSIGNEE_SEARCH_ASSIGNEE_NAME");
+
+		WorkflowMetrics.viewWorkloadByAssignee(
+			assigneeName = "3user user3",
+			isNotPresent = "4user user4,5user user5,6user user6",
+			overdueTaskCount = "0",
+			onTimeTaskCount = "0",
+			totalPendingTaskCount = "1");
+
+		WorkflowMetrics.selectWorkloadByAssigneeCardFilterValue(workflowStep = "4Role");
+
+		Click(locator1 = "WorkflowMetrics#METRICS_WORKLOAD_BY_ASSIGNEE_SEARCH_ASSIGNEE_NAME");
+
+		WorkflowMetrics.viewWorkloadByAssignee(
+			assigneeName = "4user user4",
+			isNotPresent = "5user user5,6user user6",
+			overdueTaskCount = "0",
+			onTimeTaskCount = "0",
+			totalPendingTaskCount = "1");
+
+		WorkflowMetrics.selectWorkloadByAssigneeCardFilterValue(workflowStep = "5Role");
+
+		Click(locator1 = "WorkflowMetrics#METRICS_WORKLOAD_BY_ASSIGNEE_SEARCH_ASSIGNEE_NAME");
+
+		WorkflowMetrics.viewWorkloadByAssignee(
+			assigneeName = "5user user5",
+			isNotPresent = "6user user6",
+			overdueTaskCount = "0",
+			onTimeTaskCount = "0",
+			totalPendingTaskCount = "1");
+
+		WorkflowMetrics.selectWorkloadByAssigneeCardFilterValue(workflowStep = "6Role");
+	}
+
```

Comment:
Hi @calvinctvo ,
I don't think we need all of these validations. If we check for 2 roles twice I think it is good enough for the filter validation's sake...